pydoctor/driver.py

```diff
@@ -316,17 +320,20 @@ def main(args=sys.argv[1:]):
             error("The system does not contain any code, did you "
                   "forget an --add-package?")
 
-        system.process()
 
         if system.options.projectname is None:
             name = '/'.join(ro.name for ro in system.rootobjects)
-            system.msg(
-                'warning',
-                'WARNING: guessing '+name+' for project name', thresh=-1)
+            system._warning(
+                current=None,
+                message='guessing {} for project name'.format(name),
+                detail='',
+                )
             system.projectname = name
         else:
             system.projectname = system.options.projectname
 
+        system.process()
```

Comment:
Why is this statement moved? The initialization is still quite fragile, so changing the order of things should be done very carefully and only when needed.

Reply:
It was just moved to have the project name during processing.

I think that we should not be afraid of developing pydoctor... if something breaks, an error is reported and fixed and more tests are added along the way

Reply:
I agree that fear shouldn't be a reason to not make changes. However, many attributes in the model classes are initialized at undocumented times, so I have been bitten before by changes that seemed innocent.

I do think that it makes sense to initialize as much as possible before starting the processing, so let's keep this change.